pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java

```diff
@@ -303,13 +303,12 @@ public void addFailed(ManagedLedgerException exception, Object ctx) {
         @Override
         public void run() {
             try {
-                while (lastConfirmedEntry.compareTo(currentLoadPosition) > 0) {
-                    if (((ManagedCursorImpl) cursor).isClosed()) {
-                        log.warn("[{}] MLPendingAckStore cursor have been closed, close replay thread.",
-                                cursor.getManagedLedger().getName());
-                        return;
-                    }
-                    fillEntryQueueCallback.fillQueue();
+                if (((ManagedCursorImpl) cursor).isClosed()) {
```

Comment:
this kind of casts are always a code smell and add tech debt.

can we add "isClosed" to the ManagedCursor interface ? 

otherwise if this class is meant to work only with ManagedCursorImpl then we must change the type of "cursor" to ManagedCursorImpl and to not perform this casts

Reply:
I think isClosed should be the basic function of ManagedCursor, so it is suitable to add an interface here. Thanks for your reminder, I will add this right away.